letsencrypt/cli.py

```diff
@@ -657,6 +686,7 @@ def obtain_cert(config, plugins):
 
     # This is a special case; cert and chain are simply saved
     if config.csr is not None:
+        assert lineage is None, "Did not expect a CSR with a RenewableCert"
```

Comment:
Scenarios where this could occur:
* Someone modified their renewal config file (highly unlikely)
* They provided a CSR on the command line when running `letsencrypt renew`. This is certainly possible.

Perhaps we should check for the latter case in the `renew` function and raise an error that the `renew` subcommand does not (currently) support `--csr` and point the user towards `certonly`.